src/components/filmPopups/FilmForm.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState } from "react";
+import DatePicker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+import { Container } from "../../styles/general";
+import { Bg, BgForm, Form, BtnPopupClose, BtnsWrapper, BtnReset, BtnSubmit } from "./filmPopups.styled";
+import SelectC from "../ui/Select/Select";
+import { ILabel } from "../home/Home";
+import { useAddMovieContext } from "../../context/addMovieContext";
+import { useEditMovieContext } from "../../context/editMovieContext";
+
+
+type Option = {
+  id: number;
+  type: string;
+};
+
+const _genreOptions: Option[] = [
+  { id: 1, type: 'documentary'},
+  { id: 2, type: 'comedy'},
+  { id: 3, type: 'horror'},
+  { id: 3, type: 'crime'},
+];
+
+
+const FilmPopup = ({labels}: {labels: ILabel}) => {
+  let { isAddPopupShown, setAddPopupShown } = useAddMovieContext();
+  let { isEditPopupShown, setEditPopupShown } = useEditMovieContext();
+  const [startDate, setStartDate] = useState<Date | [Date, Date] | null>(null);
+
+  const resetForm = (e: { preventDefault: () => void; }) => {
+    e.preventDefault();
+  };
+
+  const sendFilmData = (e: { preventDefault: () => void; }) => {
+    //send data
+    e.preventDefault();
+    setAddPopupShown(!isAddPopupShown);
+  };
+
+  const sendEditedFilmData = (e: { preventDefault: () => void; }) => {
+    //send edited data
+    e.preventDefault();
+    setEditPopupShown(!isEditPopupShown);
+  };
+
+  const closeAddMoviePopup = (e: { preventDefault: () => void; }) => {
+    e.preventDefault();
+    setAddPopupShown(!isAddPopupShown);
+  };
+
+  const closeEditMoviePopup = (e: { preventDefault: () => void; }) => {
+    e.preventDefault();
+    setEditPopupShown(!isEditPopupShown);
+  };
+
+  return (
+    // having issues with ts typing
+    <Bg>
+      <BgForm>
+        <Container>
+          <Form>
+            <h2>{labels.mainTitle}</h2>
+            {
+              isAddPopupShown ?
+                <BtnPopupClose onClick={closeAddMoviePopup}>&#10005;</BtnPopupClose>
+                :
+                <>
+                  <BtnPopupClose onClick={closeEditMoviePopup}>&#10005;</BtnPopupClose>
+                  <label htmlFor="movieId">
+                    Movie Id
+                    <input id="movieId" placeholder="pdb6fshan" readOnly={true}/>
+                  </label>
+                </>
+            }
+
+            <label htmlFor="title">
```

Comment:
As the other way around, you could come up with the universal `<LabelledInput />`-component that may accept the label, the `key` and the type of input as props. Then map your source data model onto those components. That would not necessarily be more efficient, but, probably, a bit DRY'er.